### PR TITLE
UI fixes for Stripe setup in Reader Revenue wizard

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

- Refactors Stripe-related code in preparation for feature expansion. 
- Fixes some UI issues

### How to test the changes in this Pull Request:

1. On `master`, remove (or temp. rename) the `newspack_stripe_data` option (to simulate a new site – you can also use a fresh new site)
2. Visit the Reader Revenue wizard and choose Stripe as the platform 
3. Go to Stripe Settings tab and observe errors being displayed: 

image

4. Observe a `PHP Warning:  Attempt to read property "webhookEndpoints" on null` warning logged 
5. Switch to this branch, reload, observe no errors and no warning logged 
6. Back to `master`! Now, save Stripe Setting without inputting any settings. 
7. Observe that after request completes, the tabs are in a limbo state and the UI changes to platform selection:

image

8. Switch to this branch, repeat, observe the error is displayed below the navigation tabs, and that the UI is not reset:

image

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/reader-revenue/stripe/class-stripe-webhooks.php [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Stripe Webhooks.
+ */
+class Stripe_Webhooks extends Stripe_Connection {
```
Comment (miguelpeixe) [inline includes/reader-revenue/stripe/class-stripe-webhooks.php:17]: Although they are related, I don't think `Stripe_Webhooks` should extend `Stripe_Connection`. What are the benefits of an instance of `Stripe_Webhooks` inheriting these methods?

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
The UI fixes are working as described.

I'm still seeing unexpected behavior while updating the API keys. After entering valid keys and saving the UI still shows the `Publishable Key and Secret Key are required to use Stripe.` warning:

image

It only clears the warning after a refresh. I'd expect the warning to clear after saving since the configuration is fixed and ready at this point.

File: includes/reader-revenue/stripe/class-stripe-webhooks.php (hunk 1 above)
Comment (adekbadek) [reply includes/reader-revenue/stripe/class-stripe-webhooks.php]: I agree, this looks weird. Changed in 327358f03944851b47e4d657817938c2b0e0762c

File: includes/reader-revenue/stripe/class-stripe-connection.php [hunk 2]
```diff
@@ -153,22 +134,24 @@ public static function update_stripe_data( $updated_stripe_data ) {
...
+		return update_option( self::STRIPE_DATA_OPTION_NAME, $validated_data );
 	}
 
 	/**
 	 * Get Stripe customer.
 	 *
 	 * @param string $customer_id Customer ID.
 	 */
-	private static function get_customer_by_id( $customer_id ) {
+	protected static function get_customer_by_id( $customer_id ) {
```
Comment (miguelpeixe) [inline includes/reader-revenue/stripe/class-stripe-connection.php:154]: Non-blocking, but these could also change back to `private` methods.

**Review by miguelpeixe — APPROVED** [review]
Thank you for revising! The UI is now clearing the warning on save.

Also tested donating and the flow doesn't appear to have any regression.

**Pull request merged**